opentelemetry/proto/profiles/v1experimental/profiles.proto

```diff
@@ -169,23 +279,280 @@ message ProfileContainer {
+  int32 link = 12;
+
+  // Timestamps associated with Sample represented in nanoseconds. These timestamps are expected
+  // to fall within the Profile's time range. [optional]
+  repeated fixed64 timestamps_unix_nano = 13;
+}
+
+// Provides additional context for a sample,
+// such as thread ID or allocation size, with optional units. [deprecated]
+message Label {
+  int32 key_strindex = 1;   // Index into string table
+
+  // At most one of the following must be present
+  int32 str_strindex = 2;   // Index into string table
+  int64 num = 3;
+
+  // Should only be present when num is present.
+  // Specifies the units of num.
+  // Use arbitrary string (for example, "requests") as a custom count unit.
+  // If no unit is specified, consumer may apply heuristic to deduce the unit.
+  // Consumers may also  interpret units like "bytes" and "kilobytes" as memory
+  // units and units like "seconds" and "nanoseconds" as time units,
+  // and apply appropriate unit conversions to these.
+  int32 num_unit_strindex = 4;  // Index into string table
+}
+
+// Indicates the semantics of the build_id field.
+enum BuildIdKind {
+  // Linker-generated build ID, stored in the ELF binary notes.
+  BUILD_ID_LINKER = 0;
+  // Build ID based on the content hash of the binary. Currently no particular
+  // hashing approach is standardized, so a given producer needs to define it
+  // themselves and thus unlike BUILD_ID_LINKER this kind of hash is producer-specific.
+  // We may choose to provide a standardized stable hash recommendation later.
+  BUILD_ID_BINARY_HASH = 1;
+}
+
+// Describes the mapping of a binary in memory, including its address range,
+// file offset, and metadata like build ID
+message Mapping {
+  // Unique nonzero id for the mapping. [deprecated]
+  uint64 id = 1;
+  // Address at which the binary (or DLL) is loaded into memory.
+  uint64 memory_start = 2;
+  // The limit of the address range occupied by this mapping.
+  uint64 memory_limit = 3;
+  // Offset in the binary that corresponds to the first mapped address.
+  uint64 file_offset = 4;
+  // The object this entry is loaded from.  This can be a filename on
+  // disk for the main binary and shared libraries, or virtual
+  // abstractions like "[vdso]".
+  int32 filename_strindex = 5;  // Index into string table
+  // A string that uniquely identifies a particular program version
+  // with high probability. E.g., for binaries generated by GNU tools,
+  // it could be the contents of the .note.gnu.build-id field.
+  int32 build_id_strindex = 6;  // Index into string table
+  // Specifies the kind of build id. See BuildIdKind enum for more details [optional]
+  BuildIdKind build_id_kind = 11;
+  // References to attributes in Profile.attribute_table. [optional]
+  repeated uint64 attributes = 12;
+  // The following fields indicate the resolution of symbolic info.
+  bool has_functions = 7;
+  bool has_filenames = 8;
+  bool has_line_numbers = 9;
+  bool has_inline_frames = 10;
```

Comment:
We can throw these away from the proto and turn them into attributes (as discussed in the SIG).

Reply:
Yes, there are a number of additional changes including these and the 0-indexed array issue that I deliberately left out of this PR to keep the scope a bit more manageable and facilitate quicker review. Because git doesn't have good history tracking when combining files, we're already seeing potential merge problems e.g. the doc_url field change (#588) would have to be reapplied manually if it get approved. I'd prefer to get the switchover done and then add further steps after.